src/dowel/csv_output.py

```diff
@@ -56,9 +65,17 @@ def record(self, data, prefix=''):
         else:
             raise ValueError('Unacceptable type.')
 
+    def csvwriter (self, filename, fieldnames,extrasaction,writeheader):
+        writer = csv.DictWriter(
+            filename,
+            fieldnames = fieldnames,
+            extrasaction = extrasaction)
+        if writeheader:
+            self._writer.writeheader()
```

Comment:
It's a little bit strange that this method uses `self._writer`, when it otherwise uses the new writer it constructed.

Reply:
Eliminated this redundant code.